src/gc.c

```diff
@@ -2832,15 +2833,15 @@ void *jl_gc_perm_alloc_nolock(size_t sz, int zero)
     if (__unlikely(sz > GC_PERM_POOL_LIMIT))
 #endif
         return zero ? calloc(1, sz) : malloc(sz);
-    sz = LLT_ALIGN(sz, JL_SMALL_BYTE_ALIGNMENT);
+    sz = LLT_ALIGN(sz, JL_HEAP_ALIGNMENT);
```

Comment:
This shouldn't be changed. (i.e. this should stay at 16 when you change the heap_alignment to 64)